sstables/index_reader.hh

```diff
@@ -526,6 +527,14 @@ private:
                     if (ex) {
                         return make_exception_future<index_list>(std::move(ex));
                     }
+                    if (_single_page_read) {
+                        // if the associated reader is forwarding despite having singular range, we prepare for that
+                        _single_page_read = false;
+                        auto moved_context = std::move(bound.context);
+                        return moved_context->close().then([this] {
```

Comment:
moved_context can go out of scope before close() completes, potential use after free

Reply:
I've added a `do_with` to prevent it in the new version